ledfx/virtuals.py

```diff
@@ -589,34 +591,27 @@ def thread_function(self):
                     and self._active_effect.is_active
                     and hasattr(self._active_effect, "pixels")
                 ):
-                    # self.assembled_frame = await self._ledfx.loop.run_in_executor(
-                    #     self._ledfx.thread_executor, self.assemble_frame
-                    # )
                     self.assembled_frame = self.assemble_frame()
                     if self.assembled_frame is not None and not self._paused:
                         if not self._config["preview_only"]:
-                            # self._ledfx.thread_executor.submit(self.flush)
-                            # await self._ledfx.loop.run_in_executor(
-                            #     self._ledfx.thread_executor, self.flush
-                            # )
                             self.flush()
 
                         self._fire_update_event()
 
-            # adjust for the frame assemble time, min allowed sleep 1 ms
-            # this will be more frame accurate on high res sleep systems
             run_time = timeit.default_timer() - start_time
-            sleep_time = max(
-                0.001, fps_to_sleep_interval(self.refresh_rate) - run_time
-            )
-            time.sleep(sleep_time)
-
-            # use an aggressive check for did we sleep against expected min clk
-            # for all high res scenarios this will be passive
-            # for unexpected high res sleep on windows scenarios it will adapt
-            pass_time = timeit.default_timer() - start_time
-            if pass_time < (self._min_time / 2):
-                time.sleep(max(0.001, self._min_time - pass_time))
+            # We sleep if we have time to kill, but we do nothing if we're behind
```

Comment:
Be very careful downnin here, there is mutating behaviour across windows and the various python versions, I remember it being a nightmare to get it to behave at all.

Reply:
this one does worry me, I would have to dig quiet a bit to unwind all the reasoning, but that gross sleep check is important